jenkins/PerfCI_Grafana_Deployment/Jenkinsfile

```diff
@@ -0,0 +1,168 @@
+                                cp -p benchmark_runner/grafana/perf/dashboard.json benchmark_runner/grafana/perf/dashboard_backup.json;
+                                PYTHONPATH=. sudo python3 benchmark_runner/grafana/update_grafana_dashboard.py --grafana_url="${GRAFANA_URL}" --grafana_api_key="${GRAFANA_API_KEY}" --grafana_json_path="${GRAFANA_JSON_PATH}" --grafana_folder_name="${GRAFANA_FOLDER_NAME}";
+                                cp -p benchmark_runner/grafana/perf/dashboard_backup.json benchmark_runner/grafana/perf/dashboard.json;
+                            '''
+                            sh 'git add benchmark_runner/grafana/perf/dashboard.json'
+                        }
+
+                        // Update main.libsonnet
+                        if (CHANGES_MAIN_LIBSONNET.toBoolean()) {
+                            sh '''
+                                cp -p benchmark_runner/grafana/perf/jsonnet/main.libsonnet benchmark_runner/grafana/perf/jsonnet/main.libsonnet_backup
+                                # Add code to update main.libsonnet here
+                            '''
+                            sh 'git add benchmark_runner/grafana/perf/jsonnet/main.libsonnet'
+                        }
+
+                        // Check if there are changes to commit
+                        def hasChanges = sh(script: 'git diff --exit-code', returnStatus: true) != 0
+
+                        if (hasChanges) {
+                            // Commit, pull, and push
+                            sh '''
+                                git commit -m "Update Grafana files"
+                                git pull https://${GIT_TOKEN}@${GIT_REPOSITORY} main
+                                git push https://${GIT_TOKEN}@${GIT_REPOSITORY} main
+                            '''
+                        } else {
+                            echo 'No changes to commit.'
+                        }
+                    }
+                } catch (Exception e) {
+                    currentBuild.result = 'FAILURE'
+                    error("Failed to execute Git commands: ${e.message}")
+                }
+            }
+        }
+    }
+
+
+
+    }
+
+    post {
+        always {
+            script {
+                sh '''
+                    if [[ "$(sudo podman images -q quay.io/ebattat/benchmark-runner 2> /dev/null)" != "" ]]; then
+                        sudo podman rmi -f $(sudo podman images -q quay.io/ebattat/benchmark-runner 2> /dev/null)
+                    fi
+                '''
+            }
+        }
+        failure {
+            script {
+                msg = "Build error for ${env.JOB_NAME} ${env.BUILD_NUMBER} (${env.BUILD_URL})"
+                emailext body: """\
+                    Jenkins job: ${env.BUILD_URL}\nSee the console output for more details:  ${env.BUILD_URL}consoleFull\n\n
+                """, subject: msg, to: "${CONTACT_EMAIL1}"
+            }
+        }
+        success {
+            echo 'TBD'
```

Comment:
Why "TBD"?

Reply:
To be defined, it's a placeholder for a successful job. I can remove it if it doesn't look good.